redis/reply.hh

```diff
@@ -0,0 +1,105 @@
+/*
+ * Copyright (C) 2019 pengjian.uestc @ gmail.com
+ */
+
+/*
+ * This file is part of Scylla.
+ *
+ * Scylla is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Scylla is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Scylla.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "bytes.hh"
+#include "seastar/core/sharded.hh"
+#include "seastar/core/shared_ptr.hh"
+#include <seastar/core/print.hh>
+#include "seastar/core/scattered_message.hh"
+#include "redis/exceptions.hh"
+
+using namespace seastar;
+
+namespace redis {
+
+class redis_message final {
+    seastar::lw_shared_ptr<scattered_message<char>> _message;
+public:
+    redis_message() = delete;
+    redis_message(const redis_message&) = delete;
+    redis_message& operator=(const redis_message&) = delete;
+    redis_message(redis_message&& o) noexcept : _message(std::move(o._message)) {}
+    redis_message(lw_shared_ptr<scattered_message<char>> m) noexcept : _message(m) {}
+    static future<redis_message> ok() {
+        auto m = make_lw_shared<scattered_message<char>> ();
+        m->append_static("+OK\r\n");
+        return make_ready_future<redis_message>(m);
+    }
+    static future<redis_message> pong() {
+        auto m = make_lw_shared<scattered_message<char>> ();
+        m->append_static("+PONG\r\n");
+        return make_ready_future<redis_message>(m);
+    }
+    static future<redis_message> err() {
+        auto m = make_lw_shared<scattered_message<char>> ();
+        m->append_static(":0\r\n");
+        return make_ready_future<redis_message>(m);
+    }
+    static future<redis_message> one() {
+        auto m = make_lw_shared<scattered_message<char>> ();
+        m->append_static(":1\r\n");
+        return make_ready_future<redis_message>(m);
+    }
+    static future<redis_message> zero() {
+        return err();
```

Comment:
Calling zero() in err() is better. BTW err message is a general message, there are many kind of error, not just return zero.

```
    static future<redis_message> zero() {
        auto m = make_lw_shared<scattered_message<char>> ();
        m->append_static(":0\r\n");
        return make_ready_future<redis_message>(m);
    }
    static future<redis_message> err() {
        return zero();
    }
```

Reply:
Yes. It's better to call zeor() in err().